Create and View User Notification Subscriptions

DyPickard opened the pull request:
For issues #172 #128 #181 and #124 

User can create notification subscriptions, data gets entered into 'Notification' table.
User can view all created notifications, data is pulled from the table.
* Right now view notification shows all notifications in the table since Session ID's are not available. Once sessions are added the code is ready to be adjusted for that.
* Recreated the database with missing notification table.
* delete function on the view notification page does not work yet, will be an issue for next sprint.

RyanPybus reviewed (approved):
Looks good dylan. I see you changed the notification table away from the attributes string format, ie. "magnitude:0-10;area:49.2827,123.1207,10000". The notification class anthony made works around this format, I think we should stick to it. I did a pr that just changes the database, tell me what you think

anthonyargatoff commented on SQLite/databaseSchema.sql:
```diff
@@ -20,6 +20,12 @@ CREATE TABLE user (
 create table notification(
     notifyid INTEGER PRIMARY KEY,
     userid Integer,
+    minMagnitude decimal(6, 4),
```
We had discussed storing these element in a string. We'll discuss the option tomorrow

DyPickard commented:
Sorry I did not realize that the 'attributes' tuple was meant to store everything as a string. We can talk about it tomorrow. I will hold off on merging this branch.

Varun2711 reviewed (approved):
looks good. But we should move the notification functionality from auth.py into the notification_routes.py file

DyPickard commented:
> looks good. But we should move the notification functionality from auth.py into the notification_routes.py file

Okay, i'll do that as well.

anthonyargatoff commented:
For the address input I used the [geocode[(https://geocode.maps.co/) API and it was quite easy. Basically we pass the address into the URL with '+' between the characters and it returns JSON with the coordinates. Ex: https://geocode.maps.co/search?q=UBC+Okanagan&api_key=6612fe31db408262351014htr8789af. I signed up and received the API key as shown in the URL, feel free to use it too.

I took the user input from the address field and used this function I wrote to return the string with spaces and commas removed:
```javascript
/** Removes spaces and commas from string. Used for geocode
 * 
 * @param {String} address String address
 * @returns Address with + in between elements
 */
function convertAddress(address){
    const parts = address.split(/[ ,]+/);
    return parts.join('+');
}
```
After this, I made an asynchronous call to the API, and then parsed the JSON data:

```javascript
const addressString = convertAddress(address);
const geocodeString = geocodeURL + addressString + '&api_key=' + geocodeAPIKey;
const geocode = await makeRequest(geocodeString);
latitude = geocode[0].lat;
longitude = geocode[0].
```
Since we are storing location as coordinates, this can be now put in the database. Maybe another nice feature would be to save the address string in the database to display it on the front end.

I don't know how much progress is left for the notifications. If this doesn't get done, no worries. But if you do feel like tackling it I could definitely help out.

DyPickard commented:
I have refactored the code:

* uses correct database
* uses dbmanager in each route
* uses attributes string instead of separate columns for each attribute
* view notifications now checks session id
* added dbmanager method for adding notification
* notification python functions moved to notification_routes.py
* removed location option from create notification and search (commented out, can be added back if functionality is added)

ready to be merged, please review
